docs/reference/mapping/types/nested.asciidoc

```diff
@@ -161,6 +165,9 @@ Nested documents can be:
 * sorted with <<nested-sorting,nested sorting>>.
 * retrieved and highlighted with <<nested-inner-hits,nested inner hits>>.
 
+TIP: When ingesting key-value pairs with a large, arbitrary set of keys, you might consider modeling each key-value pair as its own nested document with `key` and `value` fields. Instead, consider using the <<flattened>> datatype, which maps an entire object as a single field and allows for simple searches over its contents.
```

Comment:
For me this tip doesn't really fall under the topic 'interacting with `nested` documents'. It's actually a suggestion to avoid using nested documents, so I think it'd be good to keep it in the previous section where we introduce what nested objects are.

Alternatively, we could move it to the top of this page right after the initial description so it catches user's attention. I don't think it would be too distracting, since it's a short tip.

Reply:
Good point. I think that moving this tip after the initial description makes sense. That way, users can read it and potentially say, "Ah, looks like I should be using `flattened`!" instead of having to read deep into the page to discover that information.